sql/core/src/main/scala/org/apache/spark/status/api/v1/SqlResource.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.status.api.v1
+
+import java.util.Date
+import javax.ws.rs.{GET, Path, PathParam, Produces}
+import javax.ws.rs.core.MediaType
+
+import org.apache.spark.JobExecutionStatus
+import org.apache.spark.sql.execution.ui.{SQLAppStatusStore, SQLExecutionUIData}
+import org.apache.spark.ui.UIUtils
+
+@Produces(Array(MediaType.APPLICATION_JSON))
+private[v1] class SqlResource extends BaseAppResource {
+
+  @GET
+  def sqlList(): Seq[ExecutionData] = {
+    withUI { ui =>
+      val sqlStore = new SQLAppStatusStore(ui.store.store)
+
+      var executions = sqlStore.executionsList()
+        .map(exec => prepareExecutionData(exec))
+      if (executions.nonEmpty) {
+        executions = executions.sortBy(x => x.id)
+      }
+      executions
+    }
+  }
+
+  @GET
+  @Path("{executionId:\\d+}")
+  def sql(@PathParam("executionId") execId: Long): Seq[ExecutionData] = {
```

Comment:
Get of a single execution should not return a list.

Reply:
Changed to return single object and Error msg if not found